text/0118-priming-hints.md

```diff
@@ -0,0 +1,174 @@
+- offline: priming mechanisms should evaluate the given property with the given value for the purposes of priming through the component tree only if the priming is targeted at an offline use case
+
+We differentiate between these 2 scopes because online and offline priming use cases can be qualitatively different. Online priming is aimed at enhancing performance of rendering, with the assumption that further resources and data can be pulled from the server later on as necessary. Offline priming aims to prime a complete useful set of resources and data from the server up-front with no additional requests later. In general, online priming is more sensitive to performance and offline priming is more sensitive to missing scope.
+
+For our bar example, we could modify the barItem component by replacing the use of a @track member with a new @api property that uses could make use of this priming hint to guide a priming mechanism to always prime its contents:
+
+```js
+export default class BarItem extends LightningElement {
+
+    /**
+     * priming-hint: always true
+     */
+    @api
+    viewContent() {
+        return this._loadContent;
+    }
+
+    @api
+    loadContent() {
+        this._loadContent = true;
+
+        this.dispatchEvent(new CustomEvent('active'));
+    }
+
+    ...
+}
+```
+
+```html
+<!-- barItem.html -->
+<template>
+    <template if:true={viewContent}>
+        <slot></slot>
+    </template>
+</template>
+```
+
+## Other Possible Strategies
+
+Here is a brief survey of other strategies we considered for priming hints.
+
+### Contextual Import
+
+```js
+import {primingContext} from @salesforce/priming;
+...
+get loadContent {
+    return primingContext?.isPriming || _loadContent;
+}
+```
+
+Here we have the property itself explicitly import an indicator of priming status and use it to drive the value to return for the property. This feels like it would be too much work for developers and they could easily calculate the property incorrectly.
+
+### Annotation
+
+```js
+import {forcePriming} from @salesforce/priming;
+...
+
+@forcePriming
+get loadContent {
+    return _loadContent;
+}
+```
+
+Here the component opts in to the priming-specific override by using a decorator. Using decorators like this seems out-of-step with other parts of the LWC specification.
+
+### Metadata XML
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+<LightningComponentBundle xmlns="xmlns=http://soap.sforce.com/2006/04/metadata">
+    <isExposed>true</isExposed>
+    <primingHints>
+        <primingHint value="true">loadContent</primingHint>
+    </primingHints>
+</LightningComponentBundle>
+```
+
+We could express the priming overrides within metadata XML. This feels a little awkward because it's located separately from the code. A developer may miss making the corresponding change here when refactoring their component logic.
```

Comment:
The question for me here is whether we ever want to elevate these priming hints in the LWC programming model or whether it says an internal core only technology...
Using this xml would work well for internal core only but not allow us to expose this any other way.